core/eolearn/tests/test_utils/test_parsing.py

```diff
@@ -0,0 +1,142 @@
+        output_get_features=[(FeatureType.DATA, "bands")],
+        output_get_renamed_features=[(FeatureType.DATA, "bands", "new_bands")],
+        output_get_feature_specifications=[(FeatureType.DATA, "bands")],
+    ),
+    ParsingTestClass(
+        input=([(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands", "new_bands")]),
+        output_get_features=[(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands")],
+        output_get_renamed_features=[(FeatureType.DATA, "bands", "bands"), (FeatureType.MASK, "bands", "new_bands")],
+        output_get_feature_specifications=[(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands")],
+    ),
+    ParsingTestClass(
+        input=(
+            {
+                FeatureType.DATA: [("bands", "new_bands"), ("CLP", "new_CLP")],
+                FeatureType.MASK: [("bands", "new_bands"), ("CLP", "new_CLP")],
+            }
+        ),
+        output_get_features=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+        output_get_renamed_features=[
+            (FeatureType.DATA, "bands", "new_bands"),
+            (FeatureType.DATA, "CLP", "new_CLP"),
+            (FeatureType.MASK, "bands", "new_bands"),
+            (FeatureType.MASK, "CLP", "new_CLP"),
+        ],
+        output_get_feature_specifications=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+    ),
+]
+
+
+TEST_CASES_ELLIPSIS = [
+    ParsingTestClass(
+        input=({FeatureType.DATA: ..., FeatureType.MASK: [("bands", "new_bands"), ("CLP", "new_CLP")]}),
+        output_get_features=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+        output_get_renamed_features=[
+            (FeatureType.DATA, "bands", "bands"),
+            (FeatureType.DATA, "CLP", "CLP"),
+            (FeatureType.MASK, "bands", "new_bands"),
+            (FeatureType.MASK, "CLP", "new_CLP"),
+        ],
+        output_get_feature_specifications=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+    ),
+]
+
+
+@pytest.mark.parametrize("test_case", TEST_CASES_EMPTY + TEST_CASES)
+def test_FeatureParser(test_case: ParsingTestClass):
+    parser = FeatureParser(test_case.input)
+    assert parser.get_features() == test_case.output_get_features
+    assert parser.get_renamed_features() == test_case.output_get_renamed_features
+    assert parser.get_feature_specifications() == test_case.output_get_feature_specifications
+
+
+@pytest.mark.parametrize("test_case", TEST_CASES_ELLIPSIS)
+def test_FeatureParser_Error(test_case: ParsingTestClass):
```

Comment:
add a docstring describing why these tests are expected to fail

Reply:
done